rating-service/src/integration-test/java/com/modsen/rating/RatingIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+                    .then()
+                    .statusCode(HttpStatus.OK.value())
+                    .body("id", equalTo(rateId.intValue()))
+                    .body("rate", equalTo((float) rate.getRate()));
+        }
+
+        @Test
+        public void getRideById_shouldReturnNotFound() {
+            Long rateId = 999L;
+
+            RestAssuredMockMvc.given()
+                    .when()
+                    .get("/api/v1/rates/" + rateId)
+                    .then()
+                    .statusCode(HttpStatus.NOT_FOUND.value())
+                    .body("message", equalTo(ExceptionMessages.INSTANCE.rateNotFound(rateId)));
+        }
+
+        @Test
+        @Transactional
+        public void getRates_shouldReturnPagedResponse() {
+            Rate rate = RateTestEntityUtils.INSTANCE.createTestRate();
+            rateRepository.save(rate);
+
+            RestAssuredMockMvc.given()
+                    .when()
+                    .get("/api/v1/rates")
+                    .then()
+                    .statusCode(HttpStatus.OK.value())
+                    .body("list", hasSize(1))
+                    .body("totalElements", equalTo(1));
+        }
+    }
+
+    @Nested
+    class GetRatesByUser {
+
+        static Stream<Arguments> provideEndpointsAndResults() {
+            return Stream.of(
+                    Arguments.of("/api/v1/rates/from-passengers", 1, true),
+                    Arguments.of("/api/v1/rates/from-drivers", 1, false),
+                    Arguments.of("/api/v1/rates/from-passengers/1", 1, true),
+                    Arguments.of("/api/v1/rates/from-drivers/1", 1, false)
+            );
+        }
+
+        @ParameterizedTest
+        @MethodSource("provideEndpointsAndResults")
+        @Transactional
+        void getRates_shouldReturnPagedResponse(String endpoint, int expectedSize, boolean isPassenger) {
+            Rate rate = RateTestEntityUtils.INSTANCE.createTestRate();
+
+            if(!isPassenger) rate.setUserType(UserType.DRIVER);
+
+            rateRepository.save(rate);
+
+            RestAssuredMockMvc.given()
+                    .when()
+                    .get(endpoint)
+                    .then()
+                    .statusCode(HttpStatus.OK.value())
+                    .body("list", hasSize(expectedSize))
+                    .body("totalElements", equalTo(expectedSize));
+        }
+    }
+
+    @Nested
+    class AddRate {
+
+        @Test
+        @Transactional
+        public void addRate_shouldReturnCreatedRate() {
+            RequestRate requestRate = RateTestEntityUtils.INSTANCE.createTestRequestRate();
+
+            stubFor(WireMock.get(urlMatching("/api/v1/rides/" + requestRate.getRideId() + "/passenger/" + requestRate.getUserId() + "/exists"))
+                    .willReturn(aResponse()
+                            .withStatus(HttpStatus.OK.value())
+                            .withHeader("Content-Type", "application/json")
```

Comment:
Вместо строк лучше использовать константы, особенно, если они уже определены в библиотеке.
```suggestion
                            .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
```